src/Microsoft.ML.Transforms/Text/StopWordsRemoverTransform.cs

```diff
@@ -776,55 +779,89 @@ private void LoadStopWords(IHostEnvironment env, IChannel ch, ArgumentsBase load
+                }
+                _columns = columns;
             }
-            Metadata.Seal();
         }
 
-        /// <summary>
-        /// Public constructor corresponding to SignatureStopWordsRemoverTransform. It accepts arguments of type LoaderArguments,
-        /// and a separate array of columns (constructed by the caller -TextTransform- arguments).
-        /// </summary>
-        public CustomStopWordsRemoverTransform(IHostEnvironment env, LoaderArguments loaderArgs, IDataView input, OneToOneColumn[] column)
-            : base(env, RegistrationName, column, input, TestIsTextItem)
+        internal CustomStopWordsRemoverTransform(IHostEnvironment env, string stopwords,
+            string dataFile, string stopwordsColumn, IComponentFactory<IMultiStreamSource, IDataLoader> loader, params (string input, string output)[] columns) :
+         base(Contracts.CheckRef(env, nameof(env)).Register(RegistrationName), columns)
         {
-            Host.AssertNonEmpty(Infos);
-            Host.Assert(Infos.Length == Utils.Size(column));
 
-            Host.CheckValue(loaderArgs, nameof(loaderArgs));
-            Host.CheckUserArg(!string.IsNullOrWhiteSpace(loaderArgs.Stopwords) || loaderArgs.Stopword != null || !string.IsNullOrWhiteSpace(loaderArgs.DataFile),
-                nameof(loaderArgs.DataFile), "stopwords or datafile must be defined");
+        }
 
-            using (var ch = Host.Start(RegistrationName))
-            {
-                LoadStopWords(env, ch, loaderArgs, out _stopWordsMap);
-            }
-            Metadata.Seal();
+        public override void Save(ModelSaveContext ctx)
+        {
+            Host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            // *** Binary format ***
+            // <base>
+            SaveColumns(ctx);
+
+            const string dir = "Stopwords";
+            ctx.SaveSubModel(dir,
+                c =>
+                {
+                    Host.CheckValue(c, nameof(ctx));
+                    c.CheckAtModel();
+                    c.SetVersionInfo(GetStopwrodsManagerVersionInfo());
+
+                    // *** Binary format ***
+                    // int: number of stopwords
+                    // int[]: stopwords string ids
+                    Host.Assert(_stopWordsMap.Count > 0);
+                    ctx.Writer.Write(_stopWordsMap.Count);
+                    int id = 0;
+                    foreach (var nstr in _stopWordsMap)
+                    {
+                        Host.Assert(nstr.Id == id);
+                        ctx.SaveString(nstr.Value);
+                        id++;
+                    }
+
+                    ctx.SaveTextStream("Stopwords.txt", writer =>
+                    {
+                        foreach (var nstr in _stopWordsMap)
+                            writer.WriteLine("{0}\t{1}", nstr.Id, nstr.Value);
+                    });
```

Comment:
are we saving out the stop words twice?  I dont see this file being loaded back in and we are saving out the map above.